src/cursors/bytes.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{bytes_ext::BytesExt, cursors::RawCursor, error::Error, response::Response};
+use std::{
+    io::Result as IoResult,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+use tokio::io::{AsyncBufRead, AsyncRead, ReadBuf};
+
+/// Unlike [`RowCursor`] which emits rows deserialized as structures from
+/// RowBinary, this cursor emits raw bytes without deserialization.
+/// It can be iterated over using [`futures::StreamExt::next`].
+/// Additionally, if the requested format emits each row on a newline
+/// (e.g. `JSONEachRow`, `CSV`, `TSV`, etc.), the cursor can be converted into a
+/// stream of lines using [`futures::AsyncBufReadExt::lines`]`, which will allow
+/// for more convenient processing.
+// TODO: note about errors
```

Comment:
* note about downcasting to `clickhouse::Error` to get the original error
* probably we should use the more appropriate `io::ErrorKind` instead of `Other` for all errors
* note that the current implementation doesn't catch all errors (e.g. encoded inside JSON output).